repl/src/repl.rs

```diff
@@ -268,32 +268,35 @@ fn set_globals(
     }
 }
 
-fn compile_repl(vm: &Thread) -> Result<(), Box<StdError + Send + Sync>> {
+fn load_rustyline(vm: &Thread) -> vm::Result<vm::ExternModule> {
     vm.register_type::<Editor>("Editor", &[])?;
 
-    vm.define_global(
-        "rustyline",
-        record!(
-        new_editor => primitive!(1 new_editor),
-        readline => primitive!(2 readline)
-    ),
-    )?;
-    vm.define_global(
-        "repl_prim",
-        record!(
-        type_of_expr => primitive!(1 type_of_expr),
-        find_info => primitive!(1 find_info),
-        find_kind => primitive!(1 find_kind),
-        eval_line => primitive!(1 eval_line)
-    ),
-    )?;
+    vm::ExternModule::new(record!(
+            new_editor => primitive!(1 new_editor),
+            readline => primitive!(2 readline)
+        ))
+}
+
+fn load_repl(vm: &Thread) -> vm::Result<vm::ExternModule> {
+    vm::ExternModule::new(record!(
+            type_of_expr => primitive!(1 type_of_expr),
+            find_info => primitive!(1 find_info),
+            find_kind => primitive!(1 find_kind),
+            eval_line => primitive!(1 eval_line)
+        ))
+}
+
+fn compile_repl(vm: &Thread) -> Result<(), Box<StdError + Send + Sync>> {
+
+    add_extern_module(vm, "repl.prim" load_repl);
+    add_extern_module(vm, "rustyline", load_rustyline);
```

Comment:
I'm thinking that this should be `rustyline.prim` too, for consistency. 🤔

Reply:
I think its fine to omit the `prim` part if the "primitive" module exports the entire interface as it is meant to be used as there is no difference when writing `import!`.

Reply:
🤔 It is debatable wheter `prim` is a good name for it in any case. It is more or less an artifact of "primitive operations" being what I called things I implemented with rust functions rather than adding op codes for.

Reply:
Other options:

- `.ffi`
- `.extern`
- `.native`

Reply:
Maybe `extern` is better? It might be a bad idea though if `extern` we end up wanting to declare "`extern function signatures" (C or Rust function) in gluon code. (Might be useful as a way to declare signatures of functions provided with a dynamic library).

I am leaning towards opening an issue and punting on it. Since it does not need to be a public API and any user can name the module whatever they want. If we want to "standarize" `prim/ffi/extern/native` then we are better of doing that in conjunction with treating that module specially in some way.